tools/issue-labeler/src/IssueLabelerService/Answers/OpenAiAnswerService.cs

```diff
@@ -18,70 +19,72 @@ public OpenAiAnswerService(ILogger<AnswerFactory> logger, RepositoryConfiguratio
+
+
+            // Step 4: Deduplicate original query results as they are added
+            uniqueIssues.UnionWith(originalIssues);
+            uniqueDocs.UnionWith(originalDocs);
+
+            if (uniqueIssues.Count == 0 && uniqueDocs.Count == 0)
             {
-                throw new Exception($"Not enough relevant sources found for {issue.RepositoryName} using the Complete Triage model for issue #{issue.IssueNumber}. Documents: {docs.Count}, Issues: {issues.Count}.");
+                throw new Exception($"Not enough relevant sources found for {issue.RepositoryName} using the Complete Triage model for issue #{issue.IssueNumber}. Documents: {uniqueDocs.Count}, Issues: {uniqueIssues.Count}.");
             }
 
-            double highestScore = _ragService.GetHighestScore(issues, docs, issue.RepositoryName, issue.IssueNumber);
-            bool solution = highestScore >= solutionThreshold;
+            var highestScore = _ragService.GetHighestScore(uniqueIssues, uniqueDocs, issue.RepositoryName, issue.IssueNumber);
+            var solution = highestScore >= _serviceConfiguration.SolutionThreshold;
 
             _logger.LogInformation($"Highest relevance score among the sources: {highestScore}");
 
-            // Format issues 
-            var printableIssues = string.Join("\n\n", issues.Select(issue =>
-                $"Title: {issue.Title}\nDescription: {issue.chunk}\nURL: {issue.Url}\nScore: {issue.Score}"));
-
-            // Format documents 
-            var printableDocs = string.Join("\n\n", docs.Select(doc =>
-                $"Content: {doc.chunk}\nURL: {doc.Url}\nScore: {doc.Score}"));
-
-            var replacementsUserPrompt = new Dictionary<string, string>
-            {
-                { "Title", issue.Title },
-                { "Description", issue.Body },
-                { "PrintableDocs", printableDocs },
-                { "PrintableIssues", printableIssues }
-            };
+            var replacementsUserPrompt = BuildUserPromptData(uniqueIssues, uniqueDocs, issue);
 
             string instructions, userPrompt;
             if (solution)
             {
-                instructions = _config.SolutionInstructions;
-                userPrompt = AzureSdkIssueLabelerService.FormatTemplate(_config.SolutionUserPrompt, replacementsUserPrompt, _logger);
+                instructions = _serviceConfiguration.SolutionInstructions;
+                userPrompt = AzureSdkIssueLabelerService.FormatTemplate(_serviceConfiguration.SolutionUserPrompt, replacementsUserPrompt, _logger);
             }
             else
             {
-                instructions = _config.SuggestionInstructions;
-                userPrompt = AzureSdkIssueLabelerService.FormatTemplate(_config.SuggestionUserPrompt, replacementsUserPrompt, _logger);
+                instructions = _serviceConfiguration.SuggestionInstructions;
```

Comment:
```csharp
userPrompt = isSolution 
    ? _serviceConfiguration.SolutionUserPrompt, replacementsUserPrompt 
    : _serviceConfiguration.SuggestionUserPrompt, replacementsUserPrompt;

userPrompt = AzureSdkIssueLabelerService.FormatTemplate(userPrompt, replacementsUserPrompt, _logger);
```

Unless I'm misreading, the only difference in the if/else branches is the config parameter.   I think we can simplify a bit.